src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java

```diff
@@ -329,4 +338,24 @@ private void removeGraphValidationErrors(SubmissionEnvelope submissionEnvelope)
                         .collect(Collectors.toList())
         );
     }
+
+    public Instant getSubmissionContentLastUpdated(SubmissionEnvelope submissionEnvelope) {
+        PageRequest request = PageRequest.of(0, 1, new Sort(Sort.Direction.DESC, "updateDate"));
+        List<Project> projects = projectRepository.findBySubmissionEnvelope(submissionEnvelope, request).getContent();
+        List<Biomaterial> biomaterials = biomaterialRepository.findBySubmissionEnvelope(submissionEnvelope, request).getContent();
+        List<Protocol> protocols = protocolRepository.findBySubmissionEnvelope(submissionEnvelope, request).getContent();
+        List<Process> processes = processRepository.findBySubmissionEnvelope(submissionEnvelope, request).getContent();
+        List<File> files = fileRepository.findBySubmissionEnvelope(submissionEnvelope, request).getContent();
+        List<MetadataDocument> content = new ArrayList<>();
+
+        content.addAll(projects);
+        content.addAll(biomaterials);
+        content.addAll(protocols);
+        content.addAll(processes);
+        content.addAll(files);
+
+        Instant lastUpdateDate = content.stream().map(MetadataDocument::getUpdateDate).max(Instant::compareTo).get();
```

Comment:
```suggestion
Instant lastUpdateDate = Arrays.of(projectRepository, biomaterialRepository, protocolRepository, processRepository, fileRepository)
   .flatMap(repo -> repo.findBySubmissionEnvelope(submissionEnvelope, request))
   .collect(Collectors.toList())
   .map(MetadataDocument::getUpdateDate)
   .max(Instant::compareTo)
   .get();
```

Reply:
Looks really neat, however i tried applying it but it wouldn't work because the repositories when combined to a list will be treated as MongoRepository objects, and findBySubmissionEnvelope is not present in the parent object, So yeah, that didn't work :( I'll retain the current code.

Reply:
I like the functional style, there should be a way of making it work :)